### PR TITLE
Bug fix: POST with missing lobby response to 404

### LINKED ISSUE #308: POST /lobby/{lobbyId}/code/{player} responds with 400 for missing lobby ID
Current behavior and impact:
POST /lobby/{lobbyId}/code/{player} responds with HTTP 400 when the specified lobby ID does not exist.
The spec specifies that the Endpoint should respond with a HTTP 404

Steps to reproduce:
1. Submit code for a non existent lobby to the endpoint
2. API responds with `400: No lobby with that id`

Expected behavior:
HTTP 404 response

Acceptance Criteria:
- [ ] Implementation has been fixed

### PR DESCRIPTION
**Pull request by nikkischnelle**
Closes #308 

Changes POST to `/lobby/{lobbyId}/code/{player}` behaviour to respond with 404 if the lobby doesn't exist yet.

Will probably merge once there's one approval and no other change-requests

### CODE REVIEW

**Review by TR1NITROTOLUOL — DISMISSED** [review]
Good improvements/adjustments to the tests. Like those changes. All in all I approve!

File: backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt [hunk 1]
```diff
@@ -90,371 +89,443 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
...
-            contentType(ContentType.Application.Json)
-            setBody("{\"code\": \"someString\"}")
-        }
-
-        client.post("/api/v1/lobby/0/code/playerB") {
-            contentType(ContentType.Application.Json)
-            setBody("{\"code\": \"someOtherString\"}")
-        }
+    @Nested
+    inner class GetAndPostPlayerCode {
```
Comment (flamion) [inline backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:93]: I don't really like the naming of this class / scope.
Something like `Create and read back player code` or smth like that would be better imo

**Review by steinerthelion — APPROVED** [review]
Code looks good, changes are efficient and well-implemented

File: backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt [hunk 2]
```diff
@@ -90,371 +89,443 @@ class ShorkInterpreterControllerV1IT : AbstractControllerTest() {
...
+                client.post("/api/v1/lobby/0/join") {
+                    contentType(ContentType.Application.Json)
+                    setBody("{\"playerName\":\"playerA\"}")
+                }
+            assertEquals(HttpStatusCode.NotFound, result.status)
+        }
+    }
+
+    @Nested
+    inner class GetLobbies {
```
Comment (flamion) [inline backend/api/src/test/kotlin/software/shonk/adapters/incoming/ShorkInterpreterControllerV1IT.kt:526]: Something like GetAllLobbies would be better imo

**Review by flamion — CHANGES_REQUESTED** [review]

**Review by flamion — APPROVED** [review]

**Pull request merged**